### PR TITLE
fix: update schedule schema entry to support job ref and add a jobs entry

### PR DESCRIPTION
**Pull request by pandemicsyn**
Update's the schedule schema to support jobs and adds a shiny new jobs entry.

Should address:

-  https://github.com/meltano/meltano/issues/6170

### CODE REVIEW

**Comment by pandemicsyn** [comment]
Passes a local check-jsonschema validation, which checked all the random job and schedule permutations i've accumulated so far 😁 :

```
 meltano git:(fix-jobs-and-schedule-mel) check-jsonschema ~/projects/meltano-projects/rundev/meltano.jobs.yml --schemafile schema/meltano.schema.json
ok -- validation done
```

File: schema/meltano.schema.json [hunk 1]
```diff
@@ -762,6 +776,45 @@
...
+      ],
+      "description": "https://docs.meltano.com/reference/command-line-interface#job",
+      "additionalProperties": false,
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "This jobs unique name",
+          "examples": ["gitlab-to-jsonl"]
+        },
+        "tasks": {
```
Comment (pandemicsyn) [inline schema/meltano.schema.json:793]: this is actually lifted straight from the validation we do when adding a task via the cli: https://github.com/meltano/meltano/blob/main/src/meltano/core/task_sets.py#L17

File: schema/meltano.schema.json [hunk 2]
```diff
@@ -717,13 +724,15 @@
...
-      "required": [
-        "name",
-        "extractor",
-        "loader",
-        "transform",
-        "interval",
-        "start_date"
+      "oneOf": [
+        {"required": ["name", "job", "interval"]},
+        {"required": [
```
Comment (vicradon) [inline schema/meltano.schema.json:729]: @pandemicsyn please why does it take one of the two?

**Comment by JulesHuisman** [comment]
Looks good, thanks for picking this up!

**Review by edgarrmondragon — APPROVED** [review]

**Pull request merged**